server/datastore/mysql/hosts.go

```diff
@@ -651,6 +665,13 @@ func (ds *Datastore) EnrollHost(ctx context.Context, osqueryHostID, nodeKey stri
 				return backoff.Permanent(ctxerr.Errorf(ctx, "host identified by %s enrolling too often", osqueryHostID))
 			}
 			hostID = int64(host.ID)
+
+			if shouldCleanTeamPolicies(host.TeamID, teamID) {
```

Comment:
Looks good!, though there's one more scenario where I believe we don't need to clean up:

A global host (`host.TeamID == nil`) re-enrolling to some Team (`teamID != nil`).
Reasoning: If the host is global then `policy_membership` entries are for policies that are global (`policies.team_id IS NULL`) thus the clean up won't do anything.

Reply:
good catch, thank you. I will also rename the function parameters for the sake of clarity to `currentTeamId` `newTeamId` or similar.